pegasus/sites.v3/code.org/views/homepage_hero.haml

```diff
@@ -40,7 +43,25 @@
   - heroes_arranged.each_with_index do |entry, index|
     -# note: this gsheet display parameter unused due to IE8 fix.
     - display = entry[:active_s] == "hide" ? "none" : "block"
-    .widehero.lazyload{id: "fullwidth#{index}", "data-bg"=> "#{entry[:image_t]}", style: "background-position: #{entry[:param1_s]}"}
+    - image_2x = entry[:image_t].gsub('.jpg', '_2x.jpg')
+    - if index == 0
+      -# Use CSS media query to select hero image to preload/render first.
+      %link{rel: 'preload', href: image_2x, as: 'image', media:'(min-width: 1440px)'}
+      %link{rel: 'preload', href: entry[:image_t], as: 'image', media:'not all and (min-width: 1440px)'}
+      :css
+        @media (min-width: 1440px) {
+          #fullwidth#{index} {
+            background-image: url(#{image_2x});
+          }
+        }
+        @media not all and (min-width: 1440px) {
+          #fullwidth#{index} {
+            background-image: url(#{entry[:image_t]});
+          }
+        }
```

Comment:
1. This CSS style is dynamically generated based on server-side inputs, so can't be included in a static file.
2. `srcset` attribute is for the `img` element, whereas these images are set with a `background-image` CSS style, which is a different system that I don't think are compatible. There is an [`image-set()` CSS notation](https://drafts.csswg.org/css-images-3/#image-set-notation), but it's still just a draft with [limited support](http://caniuse.com/#feat=css-image-set), and doesn't (yet) support width-based sets (just resolution-based).